main/lsp/LSPTypechecker.cc

```diff
@@ -0,0 +1,537 @@
+#include "main/lsp/LSPTypechecker.h"
+#include "absl/synchronization/mutex.h"
+#include "absl/synchronization/notification.h"
+#include "ast/treemap/treemap.h"
+#include "common/typecase.h"
+#include "core/Unfreeze.h"
+#include "main/lsp/DefLocSaver.h"
+#include "main/lsp/LSPMessage.h"
+#include "main/lsp/LocalVarSaver.h"
+#include "main/lsp/ShowOperation.h"
+#include "main/pipeline/pipeline.h"
+
+namespace sorbet::realmain::lsp {
+using namespace std;
+
+LSPTypechecker::LSPTypechecker(const shared_ptr<spd::logger> &logger, WorkerPool &workers, LSPConfiguration config)
+    : wasShutdown(false), typecheckerThreadId(this_thread::get_id()), logger(logger), workers(workers), config(config) {
+}
+
+void LSPTypechecker::initialize(LSPFileUpdates updates, LSPOutput &output) {
+    runOnTypecheckerThread([&]() -> void {
```

Comment:
Asynchronous programming and taking arguments by reference is a trap. Let's take everything by moved copy

Reply:
I really wanted to; I actually had trouble capturing a moved copy. I compromised w/ this since it's not asynchronous _yet_ and I wanted to run tests.

I'll see if I can get it working.